### PR TITLE
Layouts, navigation, cmdK

### PR DESCRIPTION
**Pull request by GuiBibeau**
# Description / TLDR:

Proposes a layout and navigation structure to the documents. 

## Changes:

- adds CMD/Ctrl + K search
- adds responsive navigation bar and side bar
- `lint:fix` the whole repo
- adds tailwind
- adds Valtio for proxy state

## To do:

- abstract MDX pages: SSR props, Layouts

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (BelfordZ) [inline .prettierrc:1]: this is the one we use https://github.com/MetaMask/metamask-module-template/blob/main/.prettierrc.js

regarding changes to changelog, it was fixed here: https://github.com/MetaMask/metamask-module-template/commit/09cba70f2dcc776ff008e32c63cfdc85effd9551

Comment (GuiBibeau) [reply .prettierrc:1]: Thank you so much! I was struggling with linting vs prettier 😂 !

File: lib/navigation.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
+export interface NavigationItem {
+  pageName: string;
+  href: string;
+  children?: NavigationItem[];
+}
+
+export const navigation: NavigationItem[] = [
+  { pageName: 'Home', href: '/home', children: [] },
```
Comment (httpJunkie) [inline lib/navigation.ts:8]: should we set a default root route pointing at home?

**Review by httpJunkie — APPROVED** [review]
I have some suggestions that I think we might be able to include into this PR with minimal changes. They are merely suggestions and if we want to merge the PR without them we can still talk about them and decide to make these changes in another PR.

https://docs.google.com/document/d/1EaO8en7Os59f2QPJzc4u6uA3oHkZ0ry8wwTsEQHf5Ss/edit?usp=sharing

The code looks good, I do have some questions about some of the opinions, packages and code used in this PR but the questions are more to level up in NextJS myself rather than critique or criticize. Therefore I'm approving the PR but if you are interested in walking through the PR and and talking about some of the decisions, etc. it might be helpful for me or the team. 

Otherwise, if we want to go ahead merge for now and review later, that is ok with me.

File: lib/navigation.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        pageName: 'Getting Started',
+        href: '/guide/getting-started',
+      },
+      {
+        pageName: 'Ethereum Provider',
+        href: '/guide/ethereum-provider',
+      },
+    ],
+  },
+];
```
Comment (shanejonas) [inline lib/navigation.ts:24]: We should use the previous way of pulling the routes. It helps with contributors not missing pages in the sidebar/sitemap for the UI, for example this is missing the `mobile` section.

Comment (GuiBibeau) [reply lib/navigation.ts:24]: Totally with you on this! In this case, I was not done as I wanted to create a full system from pulling routes in a similar way you were doing that would dynamically create this object. 

I was planning to do so on a follow up right after

Comment (httpJunkie) [reply lib/navigation.ts:24]: Shane, is this ok to merge, considering that @GuiBibeau will be making those updates on the next PR? I will ensure that it is the next item we get to.

**Comment by GuiBibeau** [comment]
Closing temporarily to split or address concerns

**Pull request closed without merging**